routing.go

```diff
@@ -103,54 +107,33 @@ func (ps *proxyRouting) fetch(ctx context.Context, key string) (rb []byte, err e
 	}
 	defer resp.Body.Close()
 
-	// Read at most 10 KiB (max size of IPNS record).
-	rb, err = io.ReadAll(io.LimitReader(resp.Body, 10240))
 	if err != nil {
 		return nil, err
 	}
+	defer resp.Body.Close()
 
 	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("routing/get RPC returned unexpected status: %s, body: %q", resp.Status, string(rb))
-	}
-
-	parts := bytes.Split(bytes.TrimSpace(rb), []byte("\n"))
-	var b64 string
-
-	for _, part := range parts {
-		var evt routing.QueryEvent
-		err = json.Unmarshal(part, &evt)
-		if err != nil {
-			return nil, fmt.Errorf("routing/get RPC response cannot be parsed: %w", err)
-		}
-
-		if evt.Type == routing.Value {
-			b64 = evt.Extra
-			break
-		}
-	}
-
-	if b64 == "" {
-		return nil, errors.New("routing/get RPC returned no value")
+		return nil, fmt.Errorf("unexpected status from remote gateway: %s", resp.Status)
 	}
 
-	rb, err = base64.StdEncoding.DecodeString(b64)
+	rb, err := io.ReadAll(io.LimitReader(resp.Body, int64(ipns.MaxRecordSize)))
 	if err != nil {
 		return nil, err
 	}
 
-	entry, err := ipns.UnmarshalRecord(rb)
+	rec, err := ipns.UnmarshalRecord(rb)
 	if err != nil {
 		return nil, err
 	}
 
-	err = ipns.ValidateWithName(entry, name)
+	err = ipns.ValidateWithName(rec, name)
 	if err != nil {
 		return nil, err
 	}
 
 	return rb, nil
 }
 
-func (ps *proxyRouting) getRandomKuboURL() string {
-	return ps.kuboRPC[ps.rand.Intn(len(ps.kuboRPC))]
+func (ps *proxyRouting) getRandomGatewayURL() string {
+	return strings.TrimSuffix(ps.gatewayURLs[ps.rand.Intn(len(ps.gatewayURLs))], "/")
```

Comment:
:warning:  something is off,  got panic:

```
2023-08-14T22:02:19.294+0200	DEBUG	namesys	namesys/routing.go:58	RoutingResolver resolving k51qzi5uqu5dgutdk6i1ynyzgkqngpha5xpgia3a5qqp4jsh0u4csozksxel2r
panic: invalid argument to Intn

goroutine 772 [running]:
math/rand.(*Rand).Intn(0x0?, 0x0?)
	/usr/lib/go/src/math/rand/rand.go:180 +0x4c
main.(*proxyRouting).getRandomGatewayURL(0xc0000ee990)
	/home/lidel/project/ipfs/bifrost-gateway/routing.go:138 +0x27
main.(*proxyRouting).fetch(0xc0000ee990, {0x13fbbc0, 0xc000212a80}, {{0xc002201bf0?, 0x4472dc?, 0x0?}})
	/home/lidel/project/ipfs/bifrost-gateway/routing.go:90 +0x72
main.(*proxyRouting).SearchValue.func1()
	/home/lidel/project/ipfs/bifrost-gateway/routing.go:77 +0x34
created by main.(*proxyRouting).SearchValue in goroutine 770
	/home/lidel/project/ipfs/bifrost-gateway/routing.go:76 +0x15d
exit status 2
```

Reply:
@lidel the problem there is that you haven't set anything for the routing: no Kubo RPC, and no Gateways. I pushed a commit that will error instead when starting if there's no routing.

Please note that we removed the default kubo RPC: https://github.com/ipfs/bifrost-gateway/pull/185#discussion_r1289266069 which is the root cause of this.